src/extension/src/Utility.py

```diff
@@ -39,7 +39,7 @@ def delete_file(self, dir_path, file, raise_if_not_found=True):
                     return True
                 except Exception as e:
                     error_msg = "Trial {0}: Could not delete file. [File={1}] [Exception={2}]".format(retry+1, file, repr(e))
-                    self.logger.log_error(error_msg)
```

Comment:
Why are we constructing an error message and not logging it? Why is the log statement removed here?

Reply:
It was adding an error log for each trial attempt, making it look like an actual error in the logs, similar to this:
2020-09-11 19:29:42> "ERROR: Trial 1: Could not read file. [File=0.status] [Location=/var/lib/waagent/Microsoft.CPlat.Core.LinuxPatchExtension-1.6.7/status/0.status] [Exception=IOError(2, 'No such file or directory')]"
2020-09-11 19:29:43> "ERROR: Trial 2: Could not read file. [File=0.status] [Location=/var/lib/waagent/Microsoft.CPlat.Core.LinuxPatchExtension-1.6.7/status/0.status] [Exception=IOError(2, 'No such file or directory')]"

So removed the individual trial logs and keeping a single error log at the end of all attempts. the error_msg set above is used in the final error log below

Reply:
Added the statements back in as warning.